torch/csrc/autograd/engine.cpp

```diff
@@ -225,21 +225,37 @@ Engine::Engine() : max_recursion_depth_(MAX_DEPTH), non_reentrant_device_thread_
 // Send shutdown tasks to all device_ready_queues_ if no backward tasks are running
 // Even though readyQueue should be empty, shutdown tasks have the highest priority
 Engine::~Engine() {
+  // Occasionally autograd threads can hang on shutdown
```

Comment:
nit: if we land this, I think we should replace Occasionally by "On some configs".
Because it is not a flakyness issue here.